test/integration/document_import_test.go

```diff
@@ -82,6 +82,11 @@ func TestCatalogFromJSON(t *testing.T) {
 				}
 
 				for _, d := range deep.Equal(a, e) {
+					// ignore errors for empty collections vs nil for select fields
+					// TODO: this is brittle, but not dangerously so. We should still find a better way to do this.
```

Comment:
It's unclear to me why this is here, it'd be good to talk this through and/or add more explanation in the comment

Reply:
we have cases where the licenses are still nil slices on the package object, something that the presenter now normalizes to an empty slice instead during presentation, but technically they are not equivalent when doing comparison testing for the serialization loop (make package -> get json -> import json -> get packages -> compare packages against original packages). This ignores a select case of the described behavior for a single field.

Reply:
Okay, interesting. So this test is effectively testing several processes — normal cataloging, JSON presenters, and the new document important logic. Just so I can better understand, with regard to the document important logic, what are we verifying that wouldn't be caught with tests whose only notable function call was `CatalogFromJSON`? It seems like we could have a JSON document test fixture, and then assert that the returned values (e.g. distro, source metadata, catalog packages) are what we expect.

Btw, these could easily be notes for enhancement after this PR. I'm just trying to better understand what's going on in this code. And I think I'm coming to the realization that the empty slice vs. nil slice issue is more of a symptom than a root issue.

For now, I think it's fine to explain in the comment that this "exception" to the equality assertion is due to everything that's happening in the "serialization loop" (like you describe above), specifically that this test scope introduces the need to account for data changes in the JSON presenter logic.

Reply:
> It seems like we could have a JSON document test fixture...

Yes. However, this would be brittle in another way. 

As the shape of the JSON document changes, we'll need to update the JSON test fixture continually. We do this for the JSON presenter since the **unit** tests should be ensuring values are correct (which we do with a diff of the json values and depend on folks updating the snapshot with a scrutinizing eye). We already have a set of [curated test cases](https://github.com/anchore/syft/blob/main/test/integration/pkg_cases_test.go) which are [guaranteed to cover all package types](https://github.com/anchore/syft/blob/main/test/integration/pkg_coverage_test.go#L94-L99), which provides a greater variety of data testing coverage than what we do in the unit tests for the JSON presenters... using these cases along with the real JSON presenter output as test input casts a much wider net for being alerted to potential import problems that may arise.

> the empty slice vs. nil slice issue is more of a symptom than a root issue...

Ack. I came to the same conclusion, and didn't want to go to the trouble of fixing that behavior since it technically isn't "wrong" and would take more time.